core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/NoOpMetricCollector.java

```diff
@@ -34,6 +36,7 @@ public String name() {
 
     @Override
     public <T> void reportMetric(SdkMetric<T> metric, T data) {
+        log.trace(() -> "Metrics reported: " + data);
```

Comment:
What is the motivation for logging the NoOpMetricCollector? What kind of data would it write?

Reply:
This is intended for debugging purpose and that's why it's `trace`